alica_engine/src/engine/BasicPlan.cpp

```diff
@@ -0,0 +1,54 @@
+#include "engine/BasicPlan.h"
+
+#include "engine/AlicaEngine.h"
+#include "engine/model/Configuration.h"
+#include "engine/PlanInterface.h"
+#include "engine/scheduler/Scheduler.h"
+
+namespace alica
+{
+
+BasicPlan::BasicPlan()
+        : _msInterval(AlicaTime::milliseconds(DEFAULT_MS_INTERVAL))
+        , _planStarted(false)
+        , _activeRunJobId(-1)
+{
+}
+
+void BasicPlan::doInit()
+{
+    _planStarted = true;
+    init();
+    _activeRunJobId =  _ae->editScheduler().schedule(std::bind(&BasicPlan::doRun, this, nullptr), getInterval());
+}
+
+void BasicPlan::doRun(void* msg)
+{
+    run(msg);
+}
+
+void BasicPlan::doTerminate()
+{
+    _ae->editScheduler().stopJob(_activeRunJobId);
+    onTermination();
```

Comment:
@bjoernschroeder  since we are starting fresh, lets name methods onInit(), onTerminate()